### PR TITLE
Refactored alerts

### PR DESCRIPTION
**Pull request by cwackerfuss**
With the number of alerts we had, I noticed there was an opportunity to refactor these out. Instead of using `useState` in App.tsx, we can use a React context provider and interact with the new `useAlert` hook to declaratively render alerts without having to worry about as much about the mount/unmount process.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -140,23 +136,22 @@ function App() {
...
+      const delayMs = REVEAL_TIME_MS * MAX_WORD_LENGTH
 
-        setTimeout(() => {
-          setSuccessAlert('')
-          setIsStatsModalOpen(true)
-        }, ALERT_TIME_MS)
-      }, REVEAL_TIME_MS * MAX_WORD_LENGTH)
+      showSuccessAlert(winMessage, {
+        delayMs,
+        onClose: () => setIsStatsModalOpen(true),
```
Comment (cwackerfuss) [inline src/App.tsx:145]: I added the `onClose` option to account for this case. The alert is closed automatically after `ALERT_TIME_MS` (by default)

**Comment by TheKnowles** [comment]
Nice job! This is an elegant solution.

**Review by abrarhayat — APPROVED** [review]
The Alert Context implemented here is a great display of the usage of react hooks!

**Review by TheKnowles — COMMENTED** [review]
Wow I did not realize this was not submitted. Sorry for that.

**Pull request merged**